pkg/config/application.go

```diff
@@ -66,6 +67,55 @@ func NewApplicationGroup(application *Application, mspConfig *mb.MSPConfig) (*cb
 	return applicationGroup, nil
 }
 
+// AddAnchorPeer adds an anchor peer to an existing channel config transaction.
+// It must add the anchor peer to an existing org and the anchor peer must not already
+// exist in the org.
+func AddAnchorPeer(config *cb.Config, orgName string, newAnchorPeer *AnchorPeer) error {
+	if _, orgExists := config.ChannelGroup.Groups["Application"].Groups[orgName]; !orgExists {
+		return fmt.Errorf("application org %s does not exist in channel config", orgName)
+	}
+
+	applicationOrgGroup := config.ChannelGroup.Groups[ApplicationGroupKey].Groups[orgName]
+
+	anchorPeersProto := &pb.AnchorPeers{}
+
+	if anchorPeerConfigValue, anchorPeersExist := applicationOrgGroup.Values[AnchorPeersKey]; anchorPeersExist {
+		// Unmarshal existing anchor peers if the config value exists
+		err := proto.Unmarshal(anchorPeerConfigValue.Value, anchorPeersProto)
+		if err != nil {
+			return fmt.Errorf("failed unmarshalling existing anchor peers: %v", err)
+		}
+	}
+
+	anchorProtos := []*pb.AnchorPeer{}
+
+	// Persist existing anchor peers if found
+	if anchorPeersProto.AnchorPeers != nil {
+		anchorProtos = anchorPeersProto.AnchorPeers
+	}
```

Comment:
Why? If it's nil, the range won't do anything. The append will allocate if necessary.